packages/std/src/ibc.rs

```diff
@@ -75,6 +75,78 @@ pub enum IbcMsg {
+    /// Note that this does not necessarily have to be a packet sent by this contract.
+    /// The fees are taken from the contract's balance immediately and locked until the packet is handled.
+    ///
+    /// # Example
+    ///
+    /// Most commonly, you will attach this message to a response right before sending a packet using
+    /// [`IbcMsg::SendPacket`] or [`IbcMsg::Transfer`].
+    ///
+    /// ```rust
+    /// # use cosmwasm_std::{IbcMsg, IbcEndpoint, IbcFee, IbcTimeout, Coin, coins, CosmosMsg, Response, Timestamp};
+    ///
+    /// let incentivize = IbcMsg::PayPacketFee {
+    ///     port_id: "transfer".to_string(),
+    ///     channel_id: "source-channel".to_string(),
+    ///     fee: IbcFee {
+    ///         recv_fee: coins(100, "token"),
+    ///         ack_fee: coins(201, "token"),
+    ///         timeout_fee: coins(200, "token"),
+    ///     },
+    ///     relayers: vec![],
+    /// };
+    /// let transfer = IbcMsg::Transfer {
+    ///     channel_id: "source-channel".to_string(),
+    ///     to_address: "receiver".to_string(),
+    ///     amount: Coin::new(100u32, "token"),
+    ///     timeout: IbcTimeout::with_timestamp(Timestamp::from_nanos(0)),
+    ///     memo: None,
+    /// };
+    ///
+    /// # #[cfg(feature = "stargate")]
+    /// let _: Response = Response::new()
+    ///     .add_message(CosmosMsg::Ibc(incentivize))
+    ///     .add_message(CosmosMsg::Ibc(transfer));
+    /// ```
+    #[cfg(feature = "cosmwasm_2_2")]
+    PayPacketFee {
+        /// The port id on the chain where the packet is sent from (this chain).
+        port_id: String,
+        /// The channel id on the chain where the packet is sent from (this chain).
+        channel_id: String,
+        fee: IbcFee,
+        /// Allowlist of relayer addresses that can receive the fee.
+        /// This is currently not implemented and *must* be empty.
+        relayers: Vec<String>,
+    },
+    /// Incentivizes the existing IBC packet with the given port, channel and sequence with a fee.
+    /// Note that this does not necessarily have to be a packet sent by this contract.
+    /// The fees are taken from the contract's balance immediately and locked until the packet is handled.
+    #[cfg(feature = "cosmwasm_2_2")]
+    PayPacketFeeAsync {
+        /// The port id on the chain where the packet is sent from (this chain).
+        port_id: String,
+        /// The channel id on the chain where the packet is sent from (this chain).
+        channel_id: String,
+        /// The sequence number of the packet that should be incentivized.
+        sequence: u64,
+        fee: IbcFee,
+        /// Allowlist of relayer addresses that can receive the fee.
+        /// This is currently not implemented and *must* be empty.
+        relayers: Vec<String>,
+    },
+}
+
+#[derive(Serialize, Deserialize, Default, Clone, Debug, PartialEq, Eq, JsonSchema)]
+pub struct IbcFee {
+    // the packet receive fee
+    pub recv_fee: Vec<Coin>,
```

Comment:
What about renaming this to `receive_fee`? Would be friendlier? To far we mostly avoided the "recv" thing for contract developers